src/content/platform-integration/macos/platform-views.md

```diff
@@ -0,0 +1,243 @@
+
+macOS uses Hybrid composition, which means that the native `NSView` is appended
+to the view hierarchy.
+
+To create a platform view on macOS, use the following instructions:
+
+## On the Dart side
+
+On the Dart side, create a `Widget` and add the build implementation, as shown
+in the following steps.
+
+In the Dart widget file, make changes similar to those 
+shown in `native_view_example.dart`:
+
+<ol>
+<li>
+
+Add the following imports:
+
+<?code-excerpt "lib/native_view_example_4.dart (import)"?>
+```dart
+import 'package:flutter/foundation.dart';
+import 'package:flutter/services.dart';
+```
+
+</li>
+
+<li>
+
+Implement a `build()` method:
+
+<?code-excerpt "lib/native_view_example_4.dart (macos-composition)"?>
+```dart
+Widget build(BuildContext context) {
+  // This is used in the platform side to register the view.
+  const String viewType = '<platform-view-type>';
+  // Pass parameters to the platform side.
+  final Map<String, dynamic> creationParams = <String, dynamic>{};
+
+  return AppKitView(
+    viewType: viewType,
+    layoutDirection: TextDirection.ltr,
+    creationParams: creationParams,
+    creationParamsCodec: const StandardMessageCodec(),
+  );
+}
+```
+
+</li>
+</ol>
+
+For more information, see the API docs for: [`AppKitView`][].
+
+[`AppKitView`]: {{site.api}}/flutter/widgets/AppKitView-class.html
+
+## On the platform side
+
+On the platform side, use Swift:
+
+{% samplecode "macos-platform-views", "Swift" %}
+{% sample "Swift" %}
+
+Implement the factory and the platform view. The `FLNativeViewFactory` creates
+the platform view, and the platform view provides a reference to the `NSView`.
+For example, `FLNativeView.swift`:
+
+```swift
+import Cocoa
+import FlutterMacOS
+
+class FLNativeViewFactory: NSObject, FlutterPlatformViewFactory {
+    private var messenger: FlutterBinaryMessenger
+
+    init(messenger: FlutterBinaryMessenger) {
+        self.messenger = messenger
+        super.init()
+    }
+
+    func create(
+        viewIdentifier viewId: Int64,
+        arguments args: Any?
+    ) -> FlutterPlatformView {
+        return FLNativeView(
+            viewIdentifier: viewId,
+            arguments: args,
+            binaryMessenger: messenger)
+    }
+
+    /// Implementing this method is only necessary when the `arguments` in `createWithFrame` is not `nil`.
+    public func createArgsCodec() -> (FlutterMessageCodec & NSObjectProtocol)? {
+          return FlutterStandardMessageCodec.sharedInstance()
+    }
+}
+
+class FLNativeView: NSObject, FlutterPlatformView {
+    private var _view: NSView
+
+    init(
+        viewIdentifier viewId: Int64,
+        arguments args: Any?,
+        binaryMessenger messenger: FlutterBinaryMessenger?
+    ) {
+        _view = NSView()
+        super.init()
+        // macOS views can be created here
+        createNativeView(view: _view)
+    }
+
+    func view() -> NSView {
+        return _view
+    }
+
+    func createNativeView(view _view: NSView){
```

Comment:
It would be good to run all this code through the Swift autoformatter so it has standard spacing, consistent indenting with [our other example code](https://docs.flutter.dev/platform-integration/platform-channels#step-6-add-a-macos-platform-specific-implementation), etc.

Reply:
Done.